### PR TITLE
transfer goedel code, update documentation and Docker CI

### PR DESCRIPTION
**Pull request by palmskog**
@Zimmi48 any chance of some help with Nix CI?

### CODE REVIEW

**Comment by Zimmi48** [comment]
Yes, I will take care of this.

**Comment by palmskog** [comment]
@Zimmi48 I know you're busy, but just another ping here since hopefully this is not too much work. WIth Nix CI done and PR merged, I can take care of repo archiving and so on.

**Comment by Zimmi48** [comment]
Yes, sorry for the delay. I was indeed planning this for today.

File: .nix/coq-overlays/goedel/default.nix [hunk 1]
```diff
@@ -17,15 +19,20 @@ with lib;
 
   propagatedBuildInputs = [ hydra-battles ];
 
+  useDune2ifVersion = monorepo_version;
```
Comment (Zimmi48) [inline .nix/coq-overlays/goedel/default.nix:22]: Should become `useDuneifVersion` if we update nixpkgs.

File: .github/workflows/nix-action-default.yml [hunk 2]
```diff
@@ -307,72 +312,6 @@ jobs:
       name: Building/fetching current CI target
       run: NIXPKGS_ALLOW_UNFREE=1 nix-build --no-out-link --argstr bundle "default"
         --argstr job "hydra-battles-single"
-  build-doc:
```
Comment (Zimmi48) [inline .github/workflows/nix-action-default.yml:310]: Should not be merged yet because I've accidentally committed this removal?

Comment (palmskog) [reply .github/workflows/nix-action-default.yml:310]: I'll leave the merging of the PR to you then....

**Review by Zimmi48 — APPROVED** [review]

**Comment by Casteran** [comment]
Thanks @palmskog and @Zimmi48 !!!

**Pull request merged**